src/lepton/jpgcoder.cc

```diff
@@ -1590,44 +1615,145 @@ class caching_ibytestream
 
 
 namespace Sirikata {
-// Memory-caching implementation of Sirikata::DecoderReader API
-class CachingDecoderReader : public DecoderReader
-{
-    std::vector<uint8_t>    input_data;     // buffer holding entire stream contents
-    size_t                  input_pos;      // current reading position in the input_data
+// Implementation of Sirikata::DecoderReader API, using the provided std::vector
+// as input data stream
+class VectorDecoderReader : public DecoderReader {
+    const std::vector<uint8_t> &input_data;     // buffer holding entire stream contents
+    size_t                      input_pos;      // current reading position in the input_data
 
 public:
-    CachingDecoderReader(DecoderReader *base_stream)
-        : input_pos(0)
-    {
-        // Read the entire input file into input_data[]
-        const size_t CHUNK_SIZE = 1024 * 1024;  // 1 MiB looks like a reasonable read chunk
-        size_t cursize = 0;
-        for(;;) {
-            input_data.resize(cursize + CHUNK_SIZE);
-            auto bytes = IOUtil::ReadFull(base_stream, input_data.data() + cursize, CHUNK_SIZE);
-            if (bytes <= 0)
-                break;
-            cursize += bytes;
-        }
-        input_data.resize(cursize);
-    }  
-
-    // Read data into buffer and return amount of bytes read.
+    VectorDecoderReader(const std::vector<uint8_t> &_input_data)
+        : input_data(_input_data), input_pos(0) {
+    }
```

Comment:
Let's put blank line between methods, functions, before `public:/private:/protected:` and  after members.